src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/PassThroughPropagator.cs

```diff
@@ -0,0 +1,59 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+
+namespace System.Diagnostics
+{
+    internal sealed class PassThroughPropagator : TextMapPropagator
+    {
+        internal static TextMapPropagator Instance { get; } = new PassThroughPropagator();
+
+        public override IReadOnlyCollection<string> Fields { get; } = LegacyPropagator.Instance.Fields;
+
+        public override void Inject(Activity? activity, object? carrier, PropagatorSetterCallback? setter)
```

Comment:
Is the `PassThroughPropagator` behavior different from calling the Legacy propagator with the parent-most activity?

```c#
while (activity?.Parent is Activity parent)
{
    activity = parent;
}

LegacyPropagator.Instance.Inject(activity, carrier, setter);
```

Reply:
There is a difference. LegacyPropagator will always uses the Activity.Id for the traceparent. While pass through propagator can use Activity.ParentId if is set to the root activity.